src/scripts/addserver.js

```diff
@@ -0,0 +1,65 @@
+define(["appSettings", "loading", "browser", "emby-linkbutton"], function(appSettings, loading, browser) {
+    "use strict";
+
+    function handleConnectionResult(page, result) {
+        loading.hide();
+        switch (result.State) {
+            case "SignedIn":
+                var apiClient = result.ApiClient;
+                Dashboard.onServerChanged(apiClient.getCurrentUserId(), apiClient.accessToken(), apiClient), Dashboard.navigate("home.html");
+                break;
+            case "ServerSignIn":
+                Dashboard.navigate("login.html?serverid=" + result.Servers[0].Id, false, "none");
+                break;
+            case "ServerSelection":
```

Comment:
is this case really needed?

Reply:
I honestly don't know, I decided to leave it in until we figure out whether or not it's required. This return value is referenced a few times, not just here.